### ISSUE TITLE
pkgrepo on ubuntu trusty does not work

### ISSUE THREAD

**Issue by pille** [saltstack/salt#19848]
today i deployed a simple state, which worked on ubuntu precise. on trusty it silently fails:
```
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/glusterfs/init.sls:
glusterfs:
  pkg.installed:
    - names:
      - glusterfs-server
      - glusterfs-client
    - require:
      - pkgrepo: glusterfs-3.6-ppa

glusterfs-3.6-ppa:
  pkgrepo.managed:
    - ppa: gluster/glusterfs-3.6


[DEBUG   ] Results of YAML rendering: 
OrderedDict([('glusterfs', OrderedDict([('pkg.installed', [OrderedDict([('names', ['glusterfs-server', 'glusterfs-client'])]), OrderedDict([('require', [OrderedDict([('pkgrepo', 'glusterfs-3.6-ppa')])])])])])), ('glusterfs-3.6-ppa', OrderedDict([('pkgrepo.managed', [OrderedDict([('ppa', 'gluster/glusterfs-3.6')])])]))])
[INFO    ] Running state [glusterfs-3.6-ppa] at time 16:46:24.674156
[INFO    ] Executing state pkgrepo.managed for glusterfs-3.6-ppa
[INFO    ] {'repo': 'ppa:gluster/glusterfs-3.6'}
[INFO    ] Completed state [glusterfs-3.6-ppa] at time 16:46:24.786185
[INFO    ] Running state [glusterfs-client] at time 16:46:24.786418
[INFO    ] Executing state pkg.installed for glusterfs-client
[INFO    ] Executing command "dpkg-query --showformat='${Status} ${Package} ${Version} ${Architecture}\n' -W" in directory '/home/pille'
[INFO    ] Package glusterfs-client is already installed.
[INFO    ] Completed state [glusterfs-client] at time 16:46:25.232028
[INFO    ] Running state [glusterfs-server] at time 16:46:25.232430
[INFO    ] Executing state pkg.installed for glusterfs-server
[INFO    ] Package glusterfs-server is already installed.
[INFO    ] Completed state [glusterfs-server] at time 16:46:25.233181
[DEBUG   ] LazyLoaded .returner
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
local:
----------
          ID: glusterfs-3.6-ppa
    Function: pkgrepo.managed
      Result: True
     Comment: Configured package repo 'glusterfs-3.6-ppa'
     Started: 16:46:24.674156
    Duration: 112.029 ms
     Changes:   
              ----------
              repo:
                  ppa:gluster/glusterfs-3.6
----------
          ID: glusterfs
    Function: pkg.installed
        Name: glusterfs-client
      Result: True
     Comment: Package glusterfs-client is already installed.
     Started: 16:46:24.786418
    Duration: 445.61 ms
     Changes:   
----------
          ID: glusterfs
    Function: pkg.installed
        Name: glusterfs-server
      Result: True
     Comment: Package glusterfs-server is already installed.
     Started: 16:46:25.232430
    Duration: 0.751 ms
     Changes:   

Summary
------------
Succeeded: 3 (changed=1)
Failed:    0
------------
Total states run:     3
```

the glusterfs that gets installed is the shipped version 3.4.
looking in /etc/apt/sources.d i can't see the addition for the PPA.
repeated runs yield the same result.

installing manually via 
```
sudo add-apt-repository ppa:gluster/glusterfs-3.6
 
 More info: https://launchpad.net/~gluster/+archive/ubuntu/glusterfs-3.6
Press [ENTER] to continue or ctrl-c to cancel adding it
```
worked, but the forced keypress seems new.

**Comment by twellspring** [comment]
I can confirm this bug on 14.04. For the glusterfs-3.6 and for other PPAs.   Trying a non-real ppa gives the same results.  So it looks like the ppa lookup is not working and sending back incorrect results.

glusterfs-3.6-ppa:
  pkgrepo.managed:
    - ppa: xgluster/xglusterfs-3.6

Results in:
    ID: glusterfs-3.6-ppa
    Function: pkgrepo.managed
    Result: True
   Comment: Configured package repo 'glusterfs-3.6-ppa'
   Started: 19:47:20.983444
    Duration: 203.542 ms
     Changes:
              ----------
              repo:
                  ppa:xgluster/xglusterfs-3.6

**Comment by Deshke** [comment]
ppa's are bugged in 14.04 with  2014.7.

**Comment by rallytime** [comment]
Thanks for the report. I think @Deshke is correct, this looks like a duplicate of both #17709 and #18302 and should be fixed in 2014.7.1. The fix is here: #17712. 

@pille I am going to close this, but if the fix above still doesn't work for you, please leave a comment and we will be glad to open this again and re-visit the issue.
